cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManagerTest.java

```diff
@@ -0,0 +1,388 @@
+
+        Map<String, Double> configuredInKafka = new HashMap<>();
+
+        // Mock the Admin client
+        Admin mockAdminClient = mock(Admin.class);
+
+        // Mock describing the current metadata version
+        mockDescribeQuotasResultPresentWithAllFields(mockAdminClient, configuredInKafka);
+
+        // Mock the Admin Client Provider
+        AdminClientProvider mockAdminClientProvider = mockAdminClientProvider(mockAdminClient);
+
+        QuotasPluginKafka quotasPluginKafka = new QuotasPluginKafkaBuilder()
+            .withConsumerByteRate(consumerByteRate)
+            .withProducerByteRate(producerByteRate)
+            .withControllerMutationRate(mutationRate)
+            .withRequestPercentage(requestPercentage)
+            .build();
+
+        Map<String, Double> expectedResult = Map.of(
+            "producer_byte_rate", (double) producerByteRate,
+            "consumer_byte_rate", (double) consumerByteRate,
+            "request_percentage", (double) requestPercentage,
+            "controller_mutation_rate", mutationRate
+        );
+
+        // Scenario with configured QuotasPluginKafka and default user quota set in Kafka (all options) -> but with different values in both configurations
+        DefaultKafkaQuotasManager.reconcileDefaultUserQuotas(Reconciliation.DUMMY_RECONCILIATION, vertx, mockAdminClientProvider, DUMMY_IDENTITY.pemTrustSet(), DUMMY_IDENTITY.pemAuthIdentity(), quotasPluginKafka)
+            .onComplete(context.succeeding(s -> {
+                assertThat(!configuredInKafka.isEmpty(), is(true));
+                assertThat(configuredInKafka.equals(expectedResult), is(true));
+                checkpoint.flag();
+            }));
+    }
+
+    @Test
+    void testConfigureDefaultUserQuotasWithQuotasSetInKafkaAndStrimziQuotasPlugin(VertxTestContext context) {
+        Checkpoint checkpoint = context.checkpoint();
+        Map<String, Double> configuredInKafka = new HashMap<>();
+
+        // Mock the Admin client
+        Admin mockAdminClient = mock(Admin.class);
+
+        // Mock describing the current metadata version
+        assertDoesNotThrow(() -> mockDescribeQuotasResultPresentWithOneField(mockAdminClient, configuredInKafka));
+
+        // Mock the Admin Client Provider
+        AdminClientProvider mockAdminClientProvider = mockAdminClientProvider(mockAdminClient);
+
+        // Scenario with QuotasPluginStrimzi and default user quota set in Kafka
+        DefaultKafkaQuotasManager.reconcileDefaultUserQuotas(Reconciliation.DUMMY_RECONCILIATION, vertx, mockAdminClientProvider, DUMMY_IDENTITY.pemTrustSet(), DUMMY_IDENTITY.pemAuthIdentity(), new QuotasPluginStrimzi())
+            .onComplete(context.succeeding(s -> {
+                assertThat(configuredInKafka.values().stream().allMatch(Objects::isNull), is(true));
+                checkpoint.flag();
+            }));
+    }
+
+    @Test
+    void testConfigureQuotasWithQuotasSetInKafkaAndNoneQuotasPlugin(VertxTestContext context) {
```

Comment:
How does this differ from the previous test?

Reply:
I think that you mean another test that is named almost the same and both checks the situation when we pass there the null as a plugin -> this one checks the situation when the default Kafka quotas are set in Kafka and we set null as a "new quotas plugin".
The other one checks when there are no default quotas set in Kafka and we set `null` as a plugin -> that it will not fail on NPE and the quotas will not be touched.

But anyway, I will change the tests overall, so maybe these two will be handled differently